include/aca_zeta_programming.h

```diff
@@ -44,13 +45,29 @@ class FWD_Info {
   {
     return ((ip_addr == other.ip_addr) && (mac_addr == other.mac_addr));
   };
+  
+  // Overload "!=" for hash operation
+  bool operator!=(const FWD_Info &other) const
+  {
+    return ((ip_addr != other.ip_addr) || (mac_addr != other.mac_addr));
+  };
 };
+
+//Implement the hash of FWD_info based on std::hash<>
+class FWD_Info_Hash {
+  public:
+  std::size_t operator()(const FWD_Info &rhs) const
+  {
+    return std::hash<string>()(rhs.ip_addr) ^ std::hash<string>()(rhs.mac_addr);
```

Comment:
I believe this algorithm may still produce the same hash value for different ip+mac combination. Would it be better if we do:

std::hash(IP string + Mac string)